iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+    cut1.emplace_front();
+    cut1.emplace_front();
+    cut1.emplace_front(1584122);
+    cut1.emplace_front(158432);
+    cut1.emplace_front(158432);
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == 1234; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(4));
+    EXPECT_THAT(dTor, Eq(0));
+    EXPECT_THAT(cut1.size(), Eq(6));
+    EXPECT_THAT(cnt, Eq(0));
+
+    auto iter = cut1.cbefore_begin();
+    EXPECT_THAT((++iter)->m_value, Eq(158432));
+    EXPECT_THAT((++iter)->m_value, Eq(158432));
+    EXPECT_THAT((++iter)->m_value, Eq(1584122));
+    EXPECT_THAT((++iter)->m_value, Eq(C_TOR_TEST_VALUE_DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(C_TOR_TEST_VALUE_DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(15842));
+}
+
+TEST_F(forward_list_test, RemoveIfOnetoEmptyList)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == 15842; });
+
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(1));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(1));
+}
+
+TEST_F(forward_list_test, RemoveIfWithFewMatches)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(2));
+    EXPECT_THAT(cut1.size(), Eq(1));
+    EXPECT_THAT(cnt, Eq(2));
+}
+
+TEST_F(forward_list_test, RemoveIfWithAllMatches)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(0));
+    EXPECT_THAT(dTor, Eq(2));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(2));
+}
+
+TEST_F(forward_list_test, RemoveIfAllFromList)
+{
+    forward_list<CTorTest, 10> cut1;
+    cut1.emplace_front(15842);
+    cut1.emplace_front();
+    cut1.emplace_front();
+
+    auto cnt = cut1.remove_if([](const CTorTest& cut) { return cut.m_value == 15842; });
+    cnt += cut1.remove_if([](const CTorTest& cut) { return cut.m_value == C_TOR_TEST_VALUE_DEFAULT_VALUE; });
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(3));
+    EXPECT_THAT(cut1.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(3));
+}
+
+TEST_F(forward_list_test, writeContentViaDereferencedIterator)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace_front((const uint64_t)i);
+    }
+
+    auto cut{sut};
+    auto iter = cut.begin();
+    CTorTest element{TESTLISTCAPACITY + 1};
```

Comment:
Would use a  number unrelated to capacity, even a magic number in the test in two places. Otherwise the reader tries to find the reason for the capacity dependency and there is none as far as I can see. Seemingly "random" magic numbers indicate test values by default in tests (at least for me).